OBSDOCS-875: Update CLI command for granting permission to configure … (#73398)
openshift/openshift-docs: modules/monitoring-granting-users-permission-to-configure-monitoring-for-user-defined-projects.adoc

[id="granting-users-permission-to-configure-monitoring-for-user-defined-projects_{context}"]
= Granting users permission to configure monitoring for user-defined projects

> As a cluster administrator, you can grant users the rights to configure and manage monitoring for user-defined projects without giving the rights to the core {product-title} monitoring components. You can achieve this by assigning the `user-workload-monitoring-config-edit` role.

Reviewer: ```suggestion
As a cluster administrator, you can assign the `user-workload-monitoring-config-edit` role to a user to grant permission to configure and manage monitoring for user-defined projects without giving them permission to configure and manage core {product-title} monitoring components.
```

> As a cluster administrator, you can assign the `user-workload-monitoring-config-edit` role to a user to grant permission to configure and manage monitoring for user-defined projects without giving them permission to configure and manage core {product-title} monitoring components.

Reviewer 2: ```suggestion
As a cluster administrator, you can assign the `user-workload-monitoring-config-edit` role to a user to grant permission to configure and manage monitoring for user-defined projects without giving them permission to configure and manage core {product-title} monitoring components. User with `user-workload-monitoring-config-edit` rolebinding have full permission for  `user-workload-monitoring-config` configmap under `openshift-user-workload-monitoring` project
```
based on the `user-workload-monitoring-config-edit` role, @machine424 do you think the suggestion is fine?
```
$ oc -n openshift-user-workload-monitoring get role user-workload-monitoring-config-edit -oyaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  creationTimestamp: "2024-03-21T00:43:51Z"
  labels:
    app.kubernetes.io/managed-by: cluster-monitoring-operator
    app.kubernetes.io/part-of: openshift-monitoring
  name: user-workload-monitoring-config-edit
  namespace: openshift-user-workload-monitoring
  resourceVersion: "18856"
  uid: 678912d7-7a4f-4f27-a69b-7952c9e268a4
rules:
- apiGroups:
  - ""
  resourceNames:
  - user-workload-monitoring-config
  resources:
  - configmaps
  verbs:
  - get
  - list
  - watch
  - patch
  - update
```

Author: Hello @juzhao we already mention this information [here in the role descriptions](https://73398--ocpdocs-pr.netlify.app/openshift-enterprise/latest/monitoring/enabling-monitoring-for-user-defined-projects#granting-users-permission-to-monitor-user-defined-projects_enabling-monitoring-for-user-defined-projects). Do we still want to repeat this information here as well?

Reviewer 2: @eromanova97 ok, then no need now
/lgtm

> As a cluster administrator, you can assign the `user-workload-monitoring-config-edit` role to a user to grant permission to configure and manage monitoring for user-defined projects without giving them permission to configure and manage core {product-title} monitoring components.

Reviewer 3: ```suggestion
As a cluster administrator, you can assign the `user-workload-monitoring-config-edit` role to a user to grant permission to configure and manage monitoring for user-defined projects. This does not give them permission to configure and manage core {product-title} monitoring components.
```

Reviewer 3: (to break up the sentence since it's long)

Author: Thank you for the tip. I will definitely break the sentence into two. I will just divide it slightly differently as I want to keep the "without giving" part. The reason is I believe that the suggestion slightly changes the meaning. Because the suggestion slightly implies the admin would want to give all the permissions, but that it not the case. So I will break up the sentence as follows:
```suggestion
As a cluster administrator, you can assign the `user-workload-monitoring-config-edit` role to a user. This grants permission to configure and manage monitoring for user-defined projects without giving them permission to configure and manage core {product-title} monitoring components.
```

.Procedure

. Assign the `user-workload-monitoring-config-edit` role to a user in the `openshift-user-workload-monitoring` project:
+
[source,terminal]
----
$ oc -n openshift-user-workload-monitoring adm policy add-role-to-user \
  user-workload-monitoring-config-edit <user> \
  --role-namespace openshift-user-workload-monitoring
----

> . Verify that the user is correctly assigned to the `user-workload-monitoring-config-edit` role by displaying their associated role binding:

Reviewer 4: The `their` may imply that each user gets their own role binding.
The same role binding can be used to bind all users to the `user-workload-monitoring-config-edit` role.
In the `oc` command above, because you didn't specify a role binding, a default one `user-workload-monitoring-config-edit` in this case, is used/created/updated.
(`oc adm policy add-role-to-user --help` for more details if you want)

```suggestion
. Verify that the user is correctly assigned to the `user-workload-monitoring-config-edit` role by displaying the associated role binding:
```
or `the related role`? or just `displaying the role binding` wdyt?

Author: Hello @machine424 thank you for the information! I did not know we could bind more users to same role binding, that is why I used the wording, so thank you for the clarification, I will keep that in mind and make the change as suggested :+1:

Author: Also because that implies that the rolebinding can have different name, I am adding it as a valuable and then adding an example command.

.Procedure

> . Assign the `user-workload-monitoring-config-edit` role to a user in the `openshift-user-workload-monitoring` project:

Reviewer 2: ```suggestion
As a cluster administrator, you can assign the `user-workload-monitoring-config-edit` role to a user to grant permission to configure and manage monitoring for user-defined projects without giving them permission to configure and manage core {product-title} monitoring components. User with `user-workload-monitoring-config-edit` rolebinding have full permissions on user-workload-monitoring-config configmap under openshift-user-workload-monitoring project 
```
that's defined in
```
$ oc -n openshift-user-workload-monitoring get role user-workload-monitoring-config-edit -oyaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  creationTimestamp: "2024-03-21T00:43:51Z"
  labels:
    app.kubernetes.io/managed-by: cluster-monitoring-operator
    app.kubernetes.io/part-of: openshift-monitoring
  name: user-workload-monitoring-config-edit
  namespace: openshift-user-workload-monitoring
  resourceVersion: "18856"
  uid: 678912d7-7a4f-4f27-a69b-7952c9e268a4
rules:
- apiGroups:
  - ""
  resourceNames:
  - user-workload-monitoring-config
  resources:
  - configmaps
  verbs:
  - get
  - list
  - watch
  - patch
  - update
 ```
@machine424 do you think the suggestion is ok?
